source/lambda/use-case-management/model/use-case.ts

```diff
@@ -296,7 +296,7 @@ export class ChatUseCaseDeploymentAdapter extends UseCase {
         );
         cfnParameters.set(CfnParameterKeys.UseCaseConfigTableName, process.env[USE_CASE_CONFIG_TABLE_NAME_ENV_VAR]!);
 
-        cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolId, process.env[USER_POOL_ID_ENV_VAR]!);
+        cfnParameters.set(CfnParameterKeys.ExistingCognitoUserPoolId, eventBody.ExistingCognitoUserPoolId  || process.env[USER_POOL_ID_ENV_VAR]!);
```

Comment:
At a glance, I don't think that setting this CfnParameter alone will be sufficient for the functionality you are looking for. Below, we set parameters for `ExistingCognitoUserPoolClient` and `CognitoDomainPrefix` based on environment variables which are set on this lambda at deployment time, and refer to resources corresponding to the User Pool created at deployment, which would differ from those required when providing your own user pool.

Have you tested if this modification allowed deployment to succeed with the desired functionality?

Reply:
That's correct - this worked properly before CognitoDomainPrefix modification as of v2.0. We need to adjust this part as CognitoDomainPrefix seems to be set based on the default user pool id. I will coordinate directly with you on this change.

Reply:
Understood. So I think that for this to be compatible, we need all 3 values to be provided in the UI (user pool id, client id, and domain prefix). Onus would be on the user to have a working client in the user pool the wish to use.